Set non-strict export as default mode

gmagogsfm opened the pull request:
Summary:
- Flip the default value of strict argument in torch.export.export from True to False
- Update test infra to cope with the change, some of them made the assumption of strict mode as default
- Disabled some tests that fail in non-strict mode

Test Plan: Sandcastle

Differential Revision: D70228628


cc @ezyang @gchanan

angelayi commented on test/export/test_export_strict.py:
```diff
@@ -12,22 +12,22 @@
 
-def mocked_non_strict_export(*args, **kwargs):
+def mocked_strict_export(*args, **kwargs):
     # If user already specified strict, don't make it non-strict
```
```suggestion
    # If user already specified strict, don't make it strict
```

angelayi commented on test/export/testing.py:
```diff
@@ -263,6 +263,12 @@ def expectedFailureNonStrict(fn):
     return fn
 
 
+# Controls tests generated in test/export/test_export_nonstrict.py
```
```suggestion
# Controls tests generated in test/export/test_export_strict.py
```

angelayi commented on test/export/test_export.py:
```diff
@@ -9933,6 +9934,7 @@ def forward(self, x):
         ).run(ep.graph_module.code)
 
     def test_replace_unbacked_with_very_large_upperbound(self):
+        strict = True
```
why is this set to True?

gmagogsfm replied:
Because the un-modded (as in not a variant like retracability) test is not in non-strict mode, however some of the test cases like this one fails in non-strict. So I temporarily forced strict mode for them.

angelayi commented on test/export/test_export.py:
```diff
@@ -6720,7 +6722,6 @@ def forward(self):
         ep = export(m, ())
         self.assertEqual(ep.graph_signature.lifted_tensor_constants, ["x"])
 
-    @testing.expectedFailureRetraceability  # T186979579
```
Why does this error disappear now? Does this mean we should also have a variant that is Retraceability w/ Strict export?

gmagogsfm replied:
Ah, because of a bug in my code, I forgot to update the strict flag in retracing export call. Changed.

gmagogsfm committed:
Set non-strict export as default mode (#148790)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/148790

- Flip the default value of strict argument in torch.export.export from True to False
- Update test infra to cope with the change, some of them made the assumption of strict mode as default
- Disabled some tests that fail in non-strict mode

**If this diff breaks your test, please add `strict=True` to your torch.export.export callsite to mitigate**. DO NOT revert this diff.

Test Plan: Sandcastle

Reviewed By: angelayi

Differential Revision: D70228628
